process-document/src/main/kotlin/com/ritense/processdocument/listener/ProcessDefinitionAssociationCaseListener.kt

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2015-2024 Ritense BV, the Netherlands.
+ *
+ * Licensed under EUPL, Version 1.2 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" basis,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ritense.processdocument.listener
+
+import com.ritense.authorization.annotation.RunWithoutAuthorization
+import com.ritense.processdocument.domain.ProcessDocumentDefinitionRequest
+import com.ritense.processdocument.service.ProcessDefinitionCaseDefinitionService
+import com.ritense.valtimo.contract.annotation.SkipComponentScan
+import com.ritense.valtimo.contract.event.CaseDefinitionCreatedEvent
+import com.ritense.valtimo.contract.event.CaseDefinitionPreDeleteEvent
+import org.springframework.context.event.EventListener
+import org.springframework.stereotype.Component
+import org.springframework.transaction.annotation.Transactional
+
+@Transactional
+@Component
+@SkipComponentScan
+class ProcessDefinitionAssociationCaseListener(
+    private val service: ProcessDefinitionCaseDefinitionService,
+) {
+
+    @RunWithoutAuthorization
+    @EventListener(CaseDefinitionCreatedEvent::class)
+    fun handleCaseDefinitionCreatedEvent(event: CaseDefinitionCreatedEvent) {
+        if (event.duplicate) {
+            service.findProcessDefinitionCaseDefinitions(event.basedOnCaseDefinitionId!!).forEach { association ->
+                service.createProcessDocumentDefinition(
+                    ProcessDocumentDefinitionRequest(
+                        processDefinitionId = association.id.processDefinitionId,
```

Comment:
This is a tricky one. The process definition id would change as well so we can't straight up copy it.